rct229/rulesets/ashrae9012019/section5/section5rule38.py

```diff
@@ -0,0 +1,251 @@
+                    climate_zone,
+                    "EXTERIOR NON-RESIDENTIAL",
+                    "SKYLIGHT",
+                    skylit_wwr=0,
+                )["solar_heat_gain_coefficient"]
+
+            if scc_skylight_roof_ratios_dict_b[SCC.SEMI_EXTERIOR]:
+                target_shgc_semiheated = table_G34_lookup(
+                    climate_zone,
+                    "SEMI-EXTERIOR",
+                    "SKYLIGHT",
+                    skylit_wwr=0.021,
+                )["solar_heat_gain_coefficient"]
+            else:
+                target_shgc_semiheated = table_G34_lookup(
+                    climate_zone,
+                    "SEMI-EXTERIOR",
+                    "SKYLIGHT",
+                    skylit_wwr=0,
+                )["solar_heat_gain_coefficient"]
+
+            return {
+                "surface_conditioning_category_dict": get_surface_conditioning_category_dict(
+                    climate_zone, building_b
+                ),
+                "target_shgc_res": target_shgc_res,
+                "target_shgc_nonres": target_shgc_nonres,
+                "target_shgc_semiheated": target_shgc_semiheated,
+            }
+
+        def list_filter(self, context_item, data=None):
+            surface_b = context_item.baseline
+            scc = data["surface_conditioning_category_dict"][surface_b["id"]]
+
+            return (
+                get_opaque_surface_type(surface_b) == OST.ROOF
+                and surface_b.get("subsurfaces")
+                and scc != SCC.UNREGULATED
+            )
+
+        class RoofRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section5Rule38.BuildingRule.RoofRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    each_rule=Section5Rule38.BuildingRule.RoofRule.SubsurfaceRule(),
+                    index_rmr="baseline",
+                    list_path="subsurfaces[*]",
+                    manual_check_required_msg=MANUAL_CHECK_MSG,
+                )
+
+            def create_data(self, context, data=None):
+                surface_b = context.baseline
+                surface_id_b = surface_b["id"]
+
+                return {"surface_id_b": surface_id_b}
+
+            class SubsurfaceRule(RuleDefinitionBase):
+                def __init__(self):
+                    super(
+                        Section5Rule38.BuildingRule.RoofRule.SubsurfaceRule, self
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, False),
+                        required_fields={
+                            "$": ["classification", "glazed_area", "opaque_area"]
+                        },
+                    )
+
+                def is_applicable(self, context, data=None):
+                    subsurface_b = context.baseline
+
+                    return (
+                        subsurface_b.get("classification")
```

Comment:
The `classification`, `glazed_area` and `opaque_area` is required fields. So in here, we don't need to use `.get`. We can directly do `subsurface_b["classification]`

Reply:
Fixed.